### ISSUE TITLE
Error adding term Operation canceled

### ISSUE THREAD

**Issue by ElaxDev** [ripose-jp/Memento#200]
I don't know why but when I try adding a word to Anki it shows me this error and my Anki becomes unresponsive for some seconds:

![image](https://github.com/ripose-jp/Memento/assets/31047609/6d048bc0-845d-43b9-8223-e6c1e5df1600)

The ankiconnect hostname and port is set correctly and mining words with yomitan in google works so the problem seems to be with Memento.

**Comment by ripose-jp** [comment]
I can't reproduce it. Is it still happening? Sometimes this might happen because JapanesePod101 is having issues.

**Comment by ElaxDev** [comment]
Yes, is still happening, I'm not using JapanesePod101, I am using a [local audio server](https://github.com/themoeway/local-audio-yomichan)

**Comment by ripose-jp** [comment]
The local audio server seems to work fine when adding to Anki for me. Although I can't seem to play the audio in Memento.

**Comment by BigBoyBarney** [comment]
> I don't know why but when I try adding a word to Anki it shows me this error and my Anki becomes unresponsive for some seconds

If you're on Linux and are using the flatpak version of Memento, you need to give it access to `/tmp`.

**Comment by ElaxDev** [comment]
I'm on windows

**Comment by ripose-jp** [comment]
I just got done testing and I'm not able to reproduce this on Windows either. I think it's something specific to your setup. Possibly a plugin or something. I'd run a test with a fresh install of Anki with AnkiConnect and the local audio server plugin to see if the issue persists. Alternatively you can try to disable additional plugins until you find one that might be causing the issue.

Sorry I'm not able to help more with this one at the moment.

**Comment by ripose-jp** [comment]
Taking a screenshot isn't too interesting since Memento takes the screenshot before ever interacting with AnkiConnect. Do you mind downgrading to v1.2.1 and trying to make a card? v1.2.2 introduced a change that sends images as base64 over JSON rather than telling AnkiConnect to go into the filesystem. That's the only thing I can think might be causing this, though I wouldn't know why since I can't reproduce it.

**Comment by ElaxDev** [comment]
Yeah, downgrading did the trick, I don't know why it didn't work earlier.

**Comment by ripose-jp** [comment]
It's very strange that I can't reproduce this on any OS. At least it seems 8c0af668e1115c1dbbd68a6d497ad722c7a294a8 is the culprit. This commit fixes other issues though, so I'd rather keep it than revert it.

It would be incredibly helpful to know if this is affecting other users **or** how to reproduce it.

Could you post a screenshot of your Memento card template and export an example card from Anki so I can recreate your setup? https://litterbox.catbox.moe/ is a good temporary file hosting service to upload the akpg to.

**Comment by ElaxDev** [comment]
Sorry, I just see your comment, here are the screenshots of my card template:

![image](https://github.com/ripose-jp/Memento/assets/31047609/1040a8c6-a24b-434b-9a10-de7e61984466)
![image](https://github.com/ripose-jp/Memento/assets/31047609/164865a2-bc02-4826-ae2b-c12266683024)

Here is an example note: https://litter.catbox.moe/m8s1rz.apkg

**Comment by ripose-jp** [comment]
I don't know what the deal is, but I was able to import your note into Anki just fine. After that though, it just soft locked and wouldn't let me do anything. After restarting Anki, I tried to open the deck, but it just kept showing the Processing... loading bar forever. I think this is an issue with your note type and Anki. I'd recommend trying to use a different model and deck to mine on v1.2.2 and seeing what happens. If everything works fine, I'd say there's some sort of bug in Anki with regard to your model or deck.

**Comment by SwayStar123** [comment]
@ripose-jp 
I am facing the same error aswell, downgrading to 1.2.1 did not help for me 
![image](https://github.com/user-attachments/assets/e97826fc-f279-40ea-8887-6f956bb65043)
![image](https://github.com/user-attachments/assets/306957bb-fdb8-4af5-9b05-8802b9653048)

I have the same exact problems. I get the error, then anki becomes unresponsive for a few seconds, after i close the error popup, the card gets added to the deck without anything missing as far as i can tell (all the main things are there atleast)

**Comment by ripose-jp** [comment]
Can you post an apkg of your note type and a picture of your Memento card template?

**Comment by SwayStar123** [comment]
@ripose-jp Do you need any other info?

**Comment by ripose-jp** [comment]
My bad. My I didn't receive an email for your last post for some reason. By your card template, I mean in Memento's Anki Settings. It's helpful to see what markers you're using.

**Comment by SwayStar123** [comment]
@ripose-jp 
![image](https://github.com/user-attachments/assets/71c1d7fd-9b85-4b16-8a53-577f48dbfb9f)
![image](https://github.com/user-attachments/assets/765b6005-d8bb-4bd6-be74-50709bd11d83)

**Comment by SwayStar123** [comment]
Also i dont know if its related to this bug but i have to wait whole 20-40 seconds before adding a new word, its very painful

**Comment by ripose-jp** [comment]
I'm not able to repeat this behavior given your note type and your exact card pattern. I think there's probably something weird going on with your Anki plugins. Can you post which ones you have installed?

>Also i dont know if its related to this bug but i have to wait whole 20-40 seconds before adding a new word, its very painful

There are several factors that contribute to how long it takes to add a note. The quickest notes will use no `{audio}`, `{audio-media}`, `{audio-context}`, or `{screenshot}` markers since these necessitate intensive operations like making HTTP requests or transcoding to work. If you are using a Pentium III and an AOL trial disc, not much can be done to speed this up.

One thing that can slow down adding notes regardless of processing power or internet speed is using JSON audio sources that aren't working. This is because each time you add a note, your JSON audio sources need to timeout before it can continue. You can find out if this applies to you by going to **Settings** → **Options** → **Audio Sources** and checking if any audio sources have type JSON. Make sure those are working or remove them entirely.

**Comment by SwayStar123** [comment]
Oh that was it. Im using this for chinese, and the default audio source is japanese pod, so i guess it was querying that, failing, hence the error. Removing the {audio} fixed it, everything is fast and no errors.

**Comment by ripose-jp** [comment]
I'm going to close this issue since I don't think any progress will get made on it.
